pkg/ingester/metrics.go

```diff
@@ -676,3 +678,69 @@ func (sm *tsdbMetrics) setRegistryForUser(userID string, registry *prometheus.Re
 func (sm *tsdbMetrics) removeRegistryForUser(userID string) {
 	sm.regs.RemoveUserRegistry(userID, false)
 }
+
+type matcherCacheMetrics struct {
+	r *prometheus.Registry
+
+	requestsTotal *prometheus.Desc
+	hitsTotal     *prometheus.Desc
+	numItems      *prometheus.Desc
+	maxItems      *prometheus.Desc
+	evicted       *prometheus.Desc
+}
+
+func newMatchCacheMetrics(r *prometheus.Registry) *matcherCacheMetrics {
```

Comment:
Do we need to move this code to a shared place so that store gateway can reuse?
Or store gateway matchers cache will be enabled at Thanos side so we don't have to expose the metric for it